### PR TITLE
✨ `patzcuaro` types

### PR DESCRIPTION
**Pull request by LuisBrime**
Added different enums, requests and types to use on `patzcuaro` and its resources.

- `FileFormat.any` added to allow the download of any type of file from `patzcuaro`.

- Added `BatchFileMetadata` base model to validate files uploaded using `patzcuaro`'s `FileBatch` resource.

- Added `FileQuery` to query files by its type.

- Added requests:
    - `FileUploadRequest` for individual file upload using `File` resource.
    - `FileBatchUploadRequest` to validate multiple file uploads using urls to `patzcuaro`:
        - `FileRequest` validates each URL file sent in a batch.

- Added `python-magic` as a dependency to calculate file extensions from their bytes.
    - `python-magic-bin` added as a **dev** dependency to allow local development. (`python-magic` alone won't work in some OS like Mac or Windows, that's the reason why `python-magic-bin` is needed for local development).

### CODE REVIEW

File: cuenca_validations/types/queries.py [hunk 1]
```diff
@@ -147,3 +148,8 @@ class EventQuery(QueryParams):
 
 class SessionQuery(QueryParams):
     type: SessionType
+
+
+class FileQuery(QueryParams):
+    type: KYCFileType
+    extension: Optional[str] = None
```
Comment (alexviquez) [inline cuenca_validations/types/queries.py:155]: remover este filtrado, creo que solo tiene sentido filtrar por type

File: cuenca_validations/types/requests.py [hunk 2]
```diff
@@ -376,3 +376,7 @@ class SessionRequest(BaseRequest):
     type: SessionType
     success_url: Optional[str] = None
     failure_url: Optional[str] = None
+
+
+class FileRequest(BaseRequest):
+    file: bytes
```
Comment (alexviquez) [inline cuenca_validations/types/requests.py:382]: user_id no se si lo ocupes acá

**Review by alexviquez — CHANGES_REQUESTED** [review]

File: cuenca_validations/types/files.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from typing import Optional
+
+from pydantic import BaseModel, HttpUrl
+
+from .enums import KYCFileType
+
+
+class BatchFileMetadata(BaseModel):
+    id: Optional[str]
```
Comment (alexviquez) [inline cuenca_validations/types/files.py:9]: porque el id es opcional?

File: cuenca_validations/types/queries.py [hunk 4]
```diff
@@ -147,3 +148,7 @@ class EventQuery(QueryParams):
 
 class SessionQuery(QueryParams):
     type: SessionType
+
+
+class FileQuery(QueryParams):
+    type: KYCFileType
```
Comment (alexviquez) [inline cuenca_validations/types/queries.py:154]: debería ser opcional el type

File: cuenca_validations/version.py [hunk 5]
```diff
@@ -1 +1 @@
-__version__ = '0.9.18'
+__version__ = '0.9.20.dev10'
```
Comment (alexviquez) [inline cuenca_validations/version.py:1]: cambiar version

File: cuenca_validations/types/requests.py [hunk 6]
```diff
@@ -399,3 +403,31 @@ class SessionRequest(BaseRequest):
...
+    user_id: str
+
+    @property
+    def file_bytes(self) -> bytes:
+        return b64.b64decode(self.file)
+
+    @property
+    def calculated_extension(self) -> str:
+        mime = from_buffer(self.file_bytes, mime=True)
+        return guess_extension(mime) or 'none'
```
Comment (alexviquez) [inline cuenca_validations/types/requests.py:422]: Esta lógica debería vivir en patzcuaro

**Review by alexviquez — CHANGES_REQUESTED** [review]
Creo que la lógica de calcular extensión y los bites debería vivir en el back

File: cuenca_validations/types/files.py (hunk 3 above)
Comment (LuisBrime) [reply cuenca_validations/types/files.py:9]: porque este `Metadata` se usa para los archivos que se reciben para subir y los que se regresan en la descarga; los primeros no traen ID cuando lo mandan en `FileBatch`

**Pull request merged**